src/components/AttachmentModal/ModalWithoutChildren.js

```diff
@@ -0,0 +1,245 @@
+                />
+            ) : (
+                Boolean(sourceForAttachmentView) &&
+                shouldLoadAttachment && (
+                    <AttachmentView
+                        containerStyles={[styles.mh5]}
+                        source={sourceForAttachmentView}
+                        isAuthTokenRequired={isAuthTokenRequired}
+                        file={file}
+                        onToggleKeyboard={updateConfirmButtonVisibility}
+                        isWorkspaceAvatar={props.isWorkspaceAvatar}
+                        fallbackSource={props.fallbackSource}
+                        isUsedInAttachmentModal
+                    />
+                )
+            )}
+        </View>
+        {/* If we have an onConfirm method show a confirmation button */}
+        {Boolean(props.onConfirm) && (
+            <SafeAreaConsumer>
+                {({safeAreaPaddingBottomStyle}) => (
+                    <Animated.View style={[StyleUtils.fade(confirmButtonFadeAnimation), safeAreaPaddingBottomStyle]}>
+                        <Button
+                            success
+                            style={[styles.buttonConfirm, props.isSmallScreenWidth ? {} : styles.attachmentButtonBigScreen]}
+                            textStyles={[styles.buttonConfirmText]}
+                            text={translate('common.send')}
+                            onPress={submitAndClose}
+                            disabled={isConfirmButtonDisabled}
+                            pressOnEnter
+                        />
+                    </Animated.View>
+                )}
+            </SafeAreaConsumer>
+        )}
+        {isAttachmentReceipt && (
+            <ConfirmModal
+                title={translate('receipt.deleteReceipt')}
+                isVisible={isDeleteReceiptConfirmModalVisible}
+                onConfirm={deleteAndCloseModal}
+                onCancel={closeConfirmModal}
+                prompt={translate('receipt.deleteConfirmation')}
+                confirmText={translate('common.delete')}
+                cancelText={translate('common.cancel')}
+                danger
+            />
+        )}
+      </Modal>
+      {!isAttachmentReceipt && (
+          <ConfirmModal
+              title={attachmentInvalidReasonTitle ? translate(attachmentInvalidReasonTitle) : ''}
+              onConfirm={closeConfirmModal}
+              onCancel={closeConfirmModal}
+              isVisible={isAttachmentInvalid}
+              prompt={attachmentInvalidReason ? translate(attachmentInvalidReason) : ''}
+              confirmText={translate('common.close')}
+              shouldShowCancelButton={false}
+          />
+      )}
+    </>
+  )
+}
+
+// Prop types and callbacks defined in `index.js` file
+ModalWithoutChildren.propTypes = undefined;
+ModalWithoutChildren.displayName = 'ModalWithoutChildren';
+
+export default compose(
+    withWindowDimensions,
```

Comment:
Use the `useWindowDimensions` hook instead plz.

Reply:
I don't thing we actually need this component anymore. It was added to reduce flickering caused by onyx but I cannot reproduce them now with it removed.